src/odfesqlodbc/es_statement.cpp

```diff
@@ -141,13 +141,52 @@ RETCODE ExecuteStatement(StatementClass *stmt, BOOL commit) {
         SC_set_Result(stmt, res);
     }
 
+    // This will commit results for SQLExecDirect and will not commit
+    // results for SQLPrepare since only metadata is required for SQLPrepare
+    if (commit) {
+        GetNextResultSet(stmt);
+    }
+
     if (!SC_get_Curres(stmt))
         SC_set_Curres(stmt, SC_get_Result(stmt));
     stmt->diag_row_count = res->recent_processed_row_count;
 
     return CleanUp();
 }
 
+SQLRETURN GetNextResultSet(StatementClass *stmt) {
+    ConnectionClass *conn = SC_get_conn(stmt);
+    QResultClass *q_res = SC_get_Result(stmt);
+    if ((q_res == NULL) && (conn == NULL)) {
+        return SQL_ERROR;
+    }
+
+    SQLSMALLINT total_columns = -1;
+    if (!SQL_SUCCEEDED(SQLNumResultCols(stmt, &total_columns))
```

Comment:
Unless there is a stylistic reason to adhere to, please put && / || at the end of the preceding line.